lib/schemaUtils.js

```diff
@@ -1293,10 +1296,72 @@ module.exports = {
 
       // stop searching for helper if valid auth scheme is found
       if (!_.isEmpty(helper)) {
+        if (_.isEmpty(authHelper)) {
+          authHelper = helper;
+        }
+        else if (helper.type === 'apikey') {
+          authHelper.extraAPIKeys = !authHelper.extraAPIKeys ? [] : authHelper.extraAPIKeys;
+          authHelper.extraAPIKeys.push(helper);
+        }
+      }
+
+      if (!_.isEmpty(authHelper) && !shouldCheckForMultipleKeys) {
+        // Already got a security schema no need to check further
         return false;
       }
     });
-    return helper;
+    return authHelper;
+  },
+
+  getExtraAPIKeyParams: function (authHelper) {
+    const extraAPIKeys = _.get(authHelper, 'extraAPIKeys', []);
+    let headers = [],
+      queryParams = [];
+
+    // Remove extraAPIKeys property
+    _.set(authHelper, 'extraAPIKeys', undefined);
+
+    extraAPIKeys.forEach((apiKey) => {
+      let keyValuePair = {
+        key: _.get(apiKey, 'apikey.0.value'),
+        value: _.get(apiKey, 'apikey.1.value')
+      };
+
+      if (_.get(apiKey, 'apikey.2.value') === 'header') {
+        headers.push(keyValuePair);
+      }
+      else {
+        queryParams.push(keyValuePair);
+      }
+    });
+
+    return {
+      headers,
+      queryParams
+    };
+  },
+
+  areMultipleAPIKeysPresent: function (openapi) {
+    const securitySet = _.get(openapi, 'security', []);
+    let count = 0;
+
+    _.forEach(securitySet, (security) => {
```

Comment:
This is not how you should iterate for multiple API keys.

Take a look at this article: there are combinations of AND and OR in case when multiple keys are mentioned.

https://swagger.io/docs/specification/authentication/api-keys/?sbsearch=API%20Keys